src/rest/connector/libs/nxos/implementation.py

```diff
@@ -156,21 +164,31 @@ def connect(self, timeout=30, port="443", protocol='https'):
         self.session = requests.Session()
         _data = json.dumps(payload)
 
-        # Connect to the device via requests
-        if protocol == 'https':
-            response = self.session.post(login_url, data=_data, timeout=timeout, verify=False)
+        for _ in range(retries):
```

Comment:
we might not want to have dependency with Genie. but actually we have this as util in genie.

https://pubhub.devnetcloud.com/media/genie-docs/docs/userguide/utils/index.html#timeout

```python
from genie.utils.timeout import Timeout
# Try up to 60 seconds, and between interval wait 10 seconds, display timeout logs
timeout = Timeout(max_time = 60, interval = 10, disable_log = False)

while timeout.iterate():
    ret = do_something(**kwargs)
    if ret is None:
        return
    # Didn't get expected result, keep trying
    timeout.sleep()

Reply:
Thanks for the pointer. The logic is a bit different so will leave as is for now.